### ISSUE TITLE
[Feature Request] Figure Caption with Reference Link

### ISSUE THREAD

**Issue by RoyiAvital** [facelessuser/pymdown-extensions#1341]
It will be great to Figure Caption extension.

Something similar to https://github.com/Evidlo/markdown_captions.
The features required:

1. Support auto generated numbering of figures.
2. Built in reference system.

It will assist greatly with technical writing with the extensions (With Sublime Text).

**Comment by facelessuser** [comment]
I do not have, at this time, any plans to add new extensions to pymdown-extensions. The extension set is mainly based on my needs for documentation. Figures have not been a need of mine for some time now. If that changes, and there is not already a suitable extension out there, then that may change in the future.

Unfortunately, pymdown-extensions is not meant to be a one-stop-shop for any and every extension, just extensions that I felt didn't already exist, and that I had needed at one point or the other. They've been made public to help others who have similar needs.

---

As for Sublime Text. If there is already a suitable extension out there, and someone was to make it available in the Sublime Text ecosystem, you could easily use it. It is even possible that you could drop the extension in your package folder and load the extension using the correct path to that extension. I have not tried it, but I believe it could work. It is just not reasonable for me to provide every known extension in Markdown Preview as a dependency.

@gir-bot remove S: triage
@gir-bot add S: wontfix

**Comment by RoyiAvital** [comment]
I totally see your point. Just thought it might be easy to integrate the other code into your code.

Is there a way to use 3rd party extensions (Which are not `pymdown` yet are Python Markdown compatible) with your MarkDown Preview?

**Comment by facelessuser** [comment]
> I totally see your point. Just thought it might be easy to integrate the other code into your code.

Yeah, pymdown-extensions is not the only repository I maintain, so I have to be picky on what I commit to maintaining. The more extensions I take on, the more I commit to maintain and support. To give an idea of what I currently maintain, you can check out the official Python packages I support:

https://pypi.org/user/facelessuser/

Offical Sublime Text plugins:

https://packagecontrol.io/search/facelessuser

And that is not covering other projects I contribute to.

> Is there a way to use 3rd party extensions (Which are not `pymdown` yet are Python Markdown compatible) with your MarkDown Preview?

Yes, assuming they are available in your Packages folder somewhere, you should be able to add them the same way as pymdownx-extensions: `pluginfolder.subfolders.filename_with_no_extension`.

**Comment by RoyiAvital** [comment]
I'm not sure I get your instructions how to add 3rd party extension.

Could we make this a test case and add documentation how to do it?

**Comment by facelessuser** [comment]
We could, but it would be more appropriate over at the Markdown Preview repo.
